src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -51,53 +38,137 @@ class F extends CopyToTextureUtils {
-      gl.clear(gl.COLOR_BUFFER_BIT);
-
-      gl.scissor(rectWidth, 0, width - rectWidth, rectHeight);
-      gl.clearColor(0.0, 1.0, 0.0, 1.0);
-      gl.clear(gl.COLOR_BUFFER_BIT);
-
-      gl.scissor(0, rectHeight, rectWidth, height - rectHeight);
-      gl.clearColor(0.0, 0.0, 1.0, 1.0);
-      gl.clear(gl.COLOR_BUFFER_BIT);
-
-      gl.scissor(rectWidth, rectHeight, width - rectWidth, height - rectHeight);
-      gl.clearColor(0.0, 0.0, 0.0, 1.0);
-      gl.clear(gl.COLOR_BUFFER_BIT);
+
+    // The rgb10a2unorm dst texture will have tiny errors when we compare actual and expectation.
+    // This is due to the convert from 8-bit to 10-bit combined with alpha value ops. So for
+    // rgb10a2unorm dst textures, we'll set alphaValue to 1.0 to test.
+    const alphaValue = isLosePrecisionDstFormat ? 1.0 : 0.6;
+    const ctx = canvasContext as CanvasRenderingContext2D | OffscreenCanvasRenderingContext2D;
+    // Red
+    ctx.fillStyle = 'rgba(255, 0, 0,' + alphaValue + ')';
+    ctx.fillRect(0, 0, rectWidth, rectHeight);
+    // Lime
+    ctx.fillStyle = 'rgba(0, 255, 0,' + alphaValue + ')';
+    ctx.fillRect(rectWidth, 0, width - rectWidth, rectHeight);
+    // Blue
+    ctx.fillStyle = 'rgba(0, 0, 255,' + alphaValue + ')';
+    ctx.fillRect(0, rectHeight, rectWidth, height - rectHeight);
+    // Black
+    if (isLosePrecisionDstFormat) {
+      ctx.fillStyle = 'rgba(255, 255, 255, 0.0)';
+    } else {
+      ctx.fillStyle = 'rgba(0, 0, 0,' + alphaValue + ')';
     }
+    ctx.fillRect(rectWidth, rectHeight, width - rectWidth, height - rectHeight);
+
+    return { canvas, canvasContext };
+  }
+
+  initGLCanvasContent({
+    canvasType,
+    contextName,
+    width,
+    height,
+    premultiplied,
+    isLosePrecisionDstFormat,
+  }: {
+    canvasType: 'onscreen' | 'offscreen';
+    contextName: 'webgl' | 'webgl2';
+    width: number;
+    height: number;
+    premultiplied: boolean;
+    isLosePrecisionDstFormat: boolean;
+  }): {
+    canvas: HTMLCanvasElement | OffscreenCanvas;
+    canvasContext: WebGLRenderingContext | WebGL2RenderingContext;
+  } {
+    const canvas = this.createCanvas(canvasType, width, height);
+    if (canvas === null) {
+      this.skip('Cannot create canvas');
+    }
+
+    let canvasContext = null;
+    canvasContext = canvas.getContext(contextName, { premultipliedAlpha: premultiplied }) as
+      | WebGLRenderingContext
+      | WebGL2RenderingContext
+      | null;
+
+    if (canvasContext === null) {
+      this.skip(canvasType + ' canvas context not available');
+    }
+
+    const rectWidth = Math.floor(width / 2);
+    const rectHeight = Math.floor(height / 2);
+
+    const gl = canvasContext as WebGLRenderingContext | WebGL2RenderingContext;
+
+    // The rgb10a2unorm dst texture will have tiny errors when we compare actual and expectation.
+    // This is due to the convert from 8-bit to 10-bit combined with alpha value ops. So for
+    // rgb10a2unorm dst textures, we'll set alphaValue to 0.0 to test.
```

Comment:
Please explain why this fixes the problem. Also, the comment says alphaValue is `0.0` but code says `1.0`.